experiment/nipopow_verifier/tools/proof/proof.py

```diff
@@ -75,3 +89,175 @@ def set(self, proof, proof_name=""):
             self.hashed_headers.append(Hash(self.proof[i][0]))
 
         self.size = len(self.proof)
+
+        (
+            self.best_level,
+            self.best_score,
+            self.levels,
+            self.scores,
+        ) = best_level_and_score(self.proof)
+
+        for level in sorted(self.levels.keys()):
+            print(
+                "Level", level, "has", self.levels[level], "blocks with score", end=""
+            )
```

Comment:
All these prints can be made logging, ideally structured logging (http://www.structlog.org/en/stable/) that can be easily disabled.

Reply:
Sounds cool, I will better have a separate feature to make use of logger at all files